### PR TITLE
E2EE and VFS

### PR DESCRIPTION
**Pull request by jospoortvliet**
Describe the relation between E2EE and VFS. Also, add a section on E2EE as I could not find that anywhere...

### CODE REVIEW

**Comment by FlexW** [comment]
Sounds good to me:) You just need to signoff your commit to please the DCO bot.

File: doc/architecture.rst
Comment (allexzander) [reply doc/architecture.rst:382]: @jospoortvliet 
Nextcloud is buil**d** -> Nextcloud is buil**t**

Comment (allexzander) [reply doc/architecture.rst:382]: @jospoortvliet
for a sub-set -> for a subset

Comment (allexzander) [reply doc/architecture.rst:382]: @jospoortvliet 

**the** system administrators -> system administrators

Comment (allexzander) [reply doc/architecture.rst:382]: @jospoortvliet 

access **to** them -> access them

Comment (allexzander) [reply doc/architecture.rst:382]: @jospoortvliet 

key is also secur**i**ly -> key is also secur**e**ly

Comment (allexzander) [reply doc/architecture.rst:382]: @jospoortvliet 

requires entering **the private** key -> requires entering **this** key

Comment (allexzander) [reply doc/architecture.rst:382]: @jospoortvliet 

of **the** Nextcloud clients will -> of Nextcloud clients will

Comment (allexzander) [reply doc/architecture.rst:382]: @jospoortvliet 

to simplify adding devices -> to simplify the process of adding a new device

File: doc/architecture.rst
Comment (allexzander) [reply doc/architecture.rst:393]: @jospoortvliet 

a **maliscious** attacker -> a **malicious** attacker

File: doc/architecture.rst
Comment (allexzander) [inline doc/architecture.rst:397]: @jospoortvliet 

Nice! I would also suggest adding something like this:

In case you still want to encrypt a large file or folder, it is recommended to compress it to a password-protected archive, and then, create a text file with a password to that archive. Then, you'd need to encrypt that password-containing file with E2EE. With this approach, you can be sure your archive that contains sensitive data is still encrypted by means of archive password protection, but, the end-to-end encryption is only applied to a small text file that can be used on your local device to extract the password-protected archive.

**Comment by jospoortvliet** [comment]
> Sounds good to me:) You just need to signoff your commit to please the DCO bot.

Do you know if there's a way to do that in the github web interface? I don't want to have to check out the entire client repo just to sign something I did entirely in the browser :cry:

File: doc/architecture.rst
Comment (jospoortvliet) [reply doc/architecture.rst:397]: > @jospoortvliet
> 
> Nice! I would also suggest adding something like this:
> 
> In case you still want to encrypt a large file or folder, it is recommended to compress it to a password-protected archive, and then, create a text file with a password to that archive. Then, you'd need to encrypt that password-containing file with E2EE. With this approach, you can be sure your archive that contains sensitive data is still encrypted by means of archive password protection, but, the end-to-end encryption is only applied to a small text file that can be used on your local device to extract the password-protected archive.

Pfff, at that point, I'd say "use a password manager" or an encrypted partition or one of a dozen other solutions. I think that saying that Nextcloud is perhaps not the right solution should get the message across, giving alternative solutions isn't really worth the space in our documentation. Hmmm. I could just say 'use password protected archives or other solutions", let me see if I can reword it a bit. Tnx!

File: doc/architecture.rst
Comment (jospoortvliet) [reply doc/architecture.rst:382]: All very good and implemented. Feel free, with this kind of wording stuff, to just commit directly - I'm good with that (but I dunno what you have internally as team rules of course)

**Comment by FlexW** [comment]
> > Sounds good to me:) You just need to signoff your commit to please the DCO bot.
> 
> Do you know if there's a way to do that in the github web interface? I don't want to have to check out the entire client repo just to sign something I did entirely in the browser

I'm not sure if this is possible via the web interface. I can do it this time for you :) Next time you can just add:
```
Signed-off-by: Random Developer <random@developer.example.org>
```
to the commit message. Then the bot is fine.

**Review by allexzander — APPROVED** [review]

File: doc/architecture.rst
Comment (er-vin) [inline doc/architecture.rst:400]: Note that this limitation is only true on Windows. Granted it's only the backend in production, but I suspect if this limitation is not made more precise now it'll get overlooked when other backends come up online.

Comment (er-vin) [reply doc/architecture.rst:400]: Also I realize after posting: they do work in a way. It's just the implicit download which doesn't work on Windows. This is what you're saying, but I suspect most people will read only the first sentence. :-)

Comment (jospoortvliet) [reply doc/architecture.rst]: You are 100% correct, saying it doesn't work really is too negative. Thanks!

**Comment by jospoortvliet** [comment]
Sorry I didn't mean to re-request review from Alex, but I was a bit too quick ;-)

Anyway, feel free to merge once or let me know if there are more concerns.

**Comment by szaimen** [comment]
@jospoortvliet Thanks for this great writing! I love it 🚀

**Review by allexzander — APPROVED** [review]

**Comment by allexzander** [comment]
LGTM @er-vin If you don't mind, am gonna squash it and merge, just wanted to wait till you check your comments after they are addressed.

**Review by er-vin — COMMENTED** [review]
Looks much better now indeed.

**Pull request merged**
